conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,40 @@
+import os
+from conan.api.output import Color
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv, Environment
+
+
+class ROSEnv(object):
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.sh file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.variables = {}
+        self._virtualbuildenv = VirtualBuildEnv(conanfile)
+        self._virtualbuildenv.basename = "conanrosenv"
+        self._virtualrunenv = VirtualRunEnv(conanfile)
+
+    def generate(self):
+        output_folder = self._conanfile.generators_folder
+        self.variables.setdefault("CMAKE_TOOLCHAIN_FILE",
+                                  os.path.join(output_folder, "conan_toolchain.cmake"))
```

Comment:
```suggestion
        self.variables.setdefault("CMAKE_TOOLCHAIN_FILE",
                                  os.path.join(self._conanfile.generators_folder, "conan_toolchain.cmake"))
```
The ``output_folder`` is a different concept in Conan, the cli arg ``--output-folder``, better avoid it

Reply:
```
conan install ...
. rosenv => ``conanbuild.sh && . conanrun.sh``
(conanbuid.sh => rosenv.sh)
. conanbuild.sh && . conanrun.sh
colcon build 
. install.setup.bash
execute app
```